### PR TITLE
Small cleanups to PKI tests

### PR DESCRIPTION
**Pull request by cbandy**
These two commits:
 - Add `Equal` methods to `pki.Certificate` and `pki.PrivateKey` so they can be used by `assert.DeepEqual` in tests.
 - Remove some Kubernetes API calls from `TestReconcileCerts`.

**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?
   - [x] Have you added automated tests?

**Type of Changes:**

 
 - [x] Testing enhancement

### CODE REVIEW

File: internal/pki/pki.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+// Equal reports whether c and other have the same value.
+func (c Certificate) Equal(other Certificate) bool {
+	return bytes.Equal(c.Certificate, other.Certificate)
+}
+
+// Equal reports whether k and other have the same value.
+func (k PrivateKey) Equal(other PrivateKey) bool {
+	if k.PrivateKey == nil || other.PrivateKey == nil {
+		return k.PrivateKey == other.PrivateKey
```
Comment (jkatz) [inline internal/pki/pki.go:28]: 🤔 Do we want to evaluate `nil == nil` to `true`?

**Review by jkatz — APPROVED** [review]

File: internal/pki/pki.go (hunk 1 above)
Comment (cbandy) [reply internal/pki/pki.go:28]: `nil` happens in the zero value for `PrivateKey`, and I think we do want any value to equal itself.

Comment (jkatz) [reply internal/pki/pki.go:28]: :+1:

File: internal/pki/pki_test.go [hunk 2]
```diff
@@ -27,6 +27,38 @@ import (
...
+	root := NewRootCertificateAuthority()
+	assert.NilError(t, root.Generate())
+	assert.DeepEqual(t, root.PrivateKey, root.PrivateKey)
+
+	assert.Assert(t, !root.PrivateKey.Equal(zero))
+	assert.Assert(t, !zero.Equal(*root.PrivateKey))
+
+	other := NewRootCertificateAuthority()
+	assert.NilError(t, other.Generate())
+	assert.Assert(t, !root.PrivateKey.Equal(*other.PrivateKey))
```
Comment (benjaminjb) [inline internal/pki/pki_test.go:59]: Could we add test cases where the Certificates / PrivateKeys were equal?

Comment (cbandy) [reply internal/pki/pki_test.go:59]: That's L32, L36, L48, and L52. I'll rearrange to show that the `Equal` method works with and without `DeepEqual`.

Without the `Equal` method, `assert.DeepEqual` tries to call [`ecdsa.PrivateKey.Equal`](https://pkg.go.dev/crypto/ecdsa#PrivateKey.Equal) which panics when the receiver is nil (our zero value).

On other PrivateKey values, without the `Equal` method, `assert.DeepEqual` fails saying "cannot handle unexported field at {*pki.PrivateKey}.marshalECPrivateKey".

Comment (benjaminjb) [reply internal/pki/pki_test.go:59]: Ah, ok, I think this comment helps explain that, so maybe this comment is all you need?

Comment (cbandy) [reply internal/pki/pki_test.go:63]: Amended in af07b40d0.

**Review by benjaminjb — APPROVED** [review]

**Pull request merged**